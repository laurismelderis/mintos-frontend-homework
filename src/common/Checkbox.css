.checkbox {
  display: inline-block;
  border: 1px solid #bababa;
  border-radius: 4px;
  width: 14px;
  height: 14px;
  padding: 0;
  margin: 0;
  font-size: 16px;
  line-height: 12px;
  text-align: center;
  font-weight: 600;
  cursor: pointer;
  user-select: none;
}
