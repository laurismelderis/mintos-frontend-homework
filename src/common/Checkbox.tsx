import React from 'react'
import './Checkbox.css'

interface CheckboxProps {
  checked?: boolean
}

const Checkbox = ({ checked = false }: CheckboxProps) => {
  return (
    <div
      className='checkbox'
      style={{
        color: checked ? '#ff0404' : '#00000000',
      }}
      role='checkbox'
    >
      x
    </div>
  )
}

export default Checkbox
