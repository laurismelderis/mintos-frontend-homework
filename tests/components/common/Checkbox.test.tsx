import { render, screen } from '@testing-library/react'
import Checkbox from '../../../src/common/Checkbox'

describe('Checkbox', () => {
  it('should render a checkbox with transparent x in it', () => {
    render(<Checkbox />)

    expect(screen.getByRole('checkbox')).toBeInTheDocument()
  })

  it('should render an empty checkbox', () => {
    render(<Checkbox />)

    expect(screen.getByText('x')).toHaveStyle({
      color: '#00000000',
    })
  })

  it('should render a checked checkbox', () => {
    render(<Checkbox checked />)

    expect(screen.getByText('x')).toHaveStyle({
      color: '#ff0404',
    })
  })
})
