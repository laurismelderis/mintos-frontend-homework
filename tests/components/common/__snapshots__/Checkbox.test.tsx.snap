// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render correctly 1`] = `
<div
  className="checkbox"
  role="checkbox"
  style={
    {
      "background": "transparent",
      "color": "#ff0404",
    }
  }
>
  x
</div>
`;
